modin/experimental/xgboost/xgboost.py

```diff
@@ -12,140 +12,22 @@
 # governing permissions and limitations under the License.
 
 
-import time
 import logging
-from threading import Thread
 from typing import Dict, Optional
 from multiprocessing import cpu_count
 
 import xgboost as xgb
-import numpy as np
-import pandas
 
 from modin.config import Engine
-from modin.developer import unwrap_row_partitions
 
 if Engine.get() == "Ray":
-    import ray
-    from ray.services import get_node_ip_address
+    from .xgboost_ray import _train, _predict
 else:
     raise ValueError("Current version supports only Ray engine as MODIN_ENGINE.")
 
 LOGGER = logging.getLogger("[modin.xgboost]")
 
 
-def _start_rabit_tracker(num_workers: int):
-    """Start Rabit tracker. The workers connect to this tracker to share
-    their results."""
-    host = get_node_ip_address()
-
-    env = {"DMLC_NUM_WORKER": num_workers}
-    rabit_tracker = xgb.RabitTracker(hostIP=host, nslave=num_workers)
-
-    # Get tracker Host + IP
-    env.update(rabit_tracker.slave_envs())
-    rabit_tracker.start(num_workers)
-
-    # Wait until context completion
-    thread = Thread(target=rabit_tracker.join)
-    thread.daemon = True
-    thread.start()
-
-    return env
-
-
-class RabitContext:
```

Comment:
why this class was moved?

Reply:
This class isn't used `xgboost.py` and applicable for implementation only

Reply:
This class and `start/stop_rabit_tracker` functions was moved into separate file `utils.py` because this functions can be used in implementation with other distribute engines.